### DISCUSSION TITLE
How to use Stencil control layer

### DISCUSSION THREAD

**Discussion by CrystalSnake** [Acly/krita-ai-diffusion#685] [Q&A] [answered]
I can't understand how to use Stencil control layer for generation. What image I shoud use for this and how it's work?

**Comment by lyonnel4** [comment, marked as answer]
Hi.
it works better if you use black shapes on white background in the layer (But it works too with colored images. I guess contrast is the most important)
the model was initially made for making beautiful QRCodes but it is used most of time for a lot of things (texts, shapes, Jesus faces, etc)
ex: I did that on a layer (the one I used for the stencil, cursor to the max)
![kritastencilTestLayer](https://github.com/Acly/krita-ai-diffusion/assets/48279234/0d6ae3a6-9c3e-4e1f-915e-44afea4264a0)
then I type : "landscape with beautiful castle" in the prompt.
I generated and the result was that: if you close a little your eyes, you should see the "test" word in the picture
![kritastencilTestResult](https://github.com/Acly/krita-ai-diffusion/assets/48279234/465af31e-39ee-45dd-b650-cd955e6bda6c)

if you want the shapes to be more visible, you may add another stencil before generating or use the resulting image for a refine (you do the same again with reducing the main strength to keep a little of what was already done)

**Reply by CrystalSnake** [reply]
Thank you for the answer. Very interesting. I would never have seen this pattern without an example.
